internal/api/mfa.go

```diff
@@ -71,14 +146,23 @@ func (a *API) EnrollFactor(w http.ResponseWriter, r *http.Request) error {
 	if session == nil || user == nil {
 		return internalServerError("A valid session and a registered user are required to enroll a factor")
 	}
-
 	params := &EnrollFactorParams{}
 	if err := retrieveRequestParams(r, params); err != nil {
 		return err
 	}
 
-	if params.FactorType != models.TOTP {
-		return badRequestError(ErrorCodeValidationFailed, "factor_type needs to be totp")
+	switch params.FactorType {
+	case models.Phone:
+		if !config.MFA.Phone.EnrollEnabled {
+			return unprocessableEntityError(ErrorCodeMFAPhoneEnrollDisabled, "MFA enroll is disabled for SMS")
+		}
+		return a.enrollPhoneFactor(w, r, params)
+	case models.TOTP:
+		if !config.MFA.TOTP.EnrollEnabled {
+			return unprocessableEntityError(ErrorCodeMFATOTPEnrollDisabled, "MFA enroll is not enabled for TOTP")
```

Comment:
These two messages should be similar, one says "disabled" the other "not enabled." Let's use common wording.